### PR TITLE
Add support for Robot Framework Lint

### PR DESCRIPTION
**Pull request by nottyo**
Support [Robot Framework](http://robotframework.org/) syntax based on [Robot Framework Lint](https://github.com/boakley/robotframework-lint/wiki) project.

Usage:
```
cmd$ pip install robotframework-lint
cmd$ rflint path/to/robot/files | tee robot.log
```

Then parse `robot.log` to warnings-plugin

### CODE REVIEW

**Comment by uhafner** [comment]
Is it not possible to have a parser for the default message format of rflint? Currently, there is no way in the warnings plugin to show an explanation of the supported command.

File: src/main/java/hudson/plugins/warnings/WarningsDescriptor.java [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import java.io.IOException;
 import java.util.Collection;
 
+import org.jenkinsci.Symbol;
```
Comment (uhafner) [inline src/main/java/hudson/plugins/warnings/WarningsDescriptor.java]: Unrelated change.

File: src/main/java/hudson/plugins/warnings/parser/RFLintParser.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package hudson.plugins.warnings.parser;
+
+import hudson.Extension;
+import hudson.plugins.analysis.util.model.Priority;
+
+import java.util.regex.Matcher;
+
+/**
+ * Created by traitanit on 3/27/2017 AD.
```
Comment (uhafner) [inline src/main/java/hudson/plugins/warnings/parser/RFLintParser.java]: Replace autogenerated doc with something useful.

File: src/test/java/hudson/plugins/warnings/parser/RFLintParserTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Locale;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Created by traitanit on 3/27/2017 AD.
```
Comment (uhafner) [inline src/test/java/hudson/plugins/warnings/parser/RFLintParserTest.java]: Replace with comment.

File: warnings.iml [hunk 4]
```diff
@@ -9,26 +9,24 @@
       <sourceFolder url="file://$MODULE_DIR$/src/test/java" isTestSource="true" />
       <sourceFolder url="file://$MODULE_DIR$/src/test/resources" type="java-test-resource" />
       <sourceFolder url="file://$MODULE_DIR$/target/generated-sources/localizer" isTestSource="false" generated="true" />
-      <excludeFolder url="file://$MODULE_DIR$/target/checkout" />
```
Comment (uhafner) [inline warnings.iml]: Unrelated change. This file should not be part of the PR.

File: src/main/java/hudson/plugins/warnings/parser/RFLintParser.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                category = "ERROR";
+                break;
+            case 'W':
+                priority = Priority.NORMAL;
+                category = "WARNING";
+                break;
+            case 'I':
+                priority = Priority.LOW;
+                category = "IGNORE";
+                break;
```
Comment (uhafner) [inline src/main/java/hudson/plugins/warnings/parser/RFLintParser.java]: default:
  // do not change priority and category

File: src/main/java/hudson/plugins/warnings/WarningsDescriptor.java (hunk 1 above)
Comment (nottyo) [inline src/main/java/hudson/plugins/warnings/WarningsDescriptor.java]: I can't build the plugin without this import.

File: src/main/java/hudson/plugins/warnings/parser/RFLintParser.java (hunk 5 above)
Comment (nottyo) [inline src/main/java/hudson/plugins/warnings/parser/RFLintParser.java]: Fixed

File: warnings.iml (hunk 4 above)
Comment (nottyo) [inline warnings.iml]: Fixed

**Comment by nottyo** [comment]
@uhafner Thanks for your comment. I've fixed the RFParser to support for default option of robotframework-lint tool.

**Comment by uhafner** [comment]
Thanks! Just the change to isLineInteresting and then the PR is ready to be merged...

**Comment by nottyo** [comment]
@uhafner I decided to override the '_parse_' method for RFLintParser. I also remove all unused code including '_isLineInteresting_' method.

**Comment by uhafner** [comment]
Yes, I did see that. But what is the reason for not using the parent class implementation? You are now duplicating a lot of code!

**Comment by nottyo** [comment]
Because I want to get the filename before parsing warning messages.
Example output of robotframework-lint 
+ dir/file1.robot
W: 5, 100: Line is too long (exceeds 100 characters) (LineTooLong)
E: 10, 0: No keyword documentation (RequireKeywordDocumentation)
+ dir/file2.robot
W: 5, 100: Line is too long (exceeds 100 characters) (LineTooLong)
E: 10, 0: No keyword documentation (RequireKeywordDocumentation)

file name comes first then warning messages.

**Comment by nottyo** [comment]
I did rebase with your commits and it's going to be ready to merge. Please review.

**Comment by uhafner** [comment]
Did you rebase? The section https://github.com/jenkinsci/warnings-plugin/pull/89/files still shows unrelated work... 
Here an explanation: https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request

**Comment by nottyo** [comment]
Yes I did rebase your commits but I'm not sure why the pull request still shows unrelated work. If you take a look into my [commit history](https://github.com/nottyo/warnings-plugin/commits/master), your commit is rebased. Could you try merging my pull request?

screen shot 2560-04-06 at 10 42 58 am

**Comment by uhafner** [comment]
I see, you did not rebase, you merged your changes. This is a different thing. This keeps the wrong commits. 

But anyway, I'll merge into a branch in my repository and remove the 2 unwanted files here.

**Comment by nottyo** [comment]
ah.. sorry about that. This is my first time to rebase commits. Thanks for your help.

**Comment by uhafner** [comment]
Thanks!!! Release is on its way...

**Comment by nottyo** [comment]
Many thanks @uhafner
